src/main/java/io/proximax/sdk/model/transaction/builder/TransactionFactory.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 ProximaX Limited. All rights reserved.
+ * Use of this source code is governed by the Apache 2.0
+ * license that can be found in the LICENSE file.
+ */
+package io.proximax.sdk.model.transaction.builder;
+
+import java.math.BigInteger;
+
+import io.proximax.sdk.FeeCalculationStrategy;
+import io.proximax.sdk.model.blockchain.NetworkType;
+import io.proximax.sdk.model.transaction.PlainMessage;
+
+/**
+ * Central factory to instantiate transaction builders for all transaction types
+ */
+public class TransactionFactory {
```

Comment:
I would rename this to ```TransactionBuilderFactory```. Otherwise - this is exactly what I had in mind. Client of the sdk just parametrizes the creation of the TransactionBuilderFactory once, stores the instance for reuse and then doesn't have to bother with generationHashes, network types, fee calculation strategies, endpoints etc. Unless he wants to. I like it.

Reply:
had the same thought but was lazy :) will rename it with next commit

Reply:
renamed in 45eb4b652c2aa326198cb84699ab275409264ea3